File: front_end/portfolio_ember_front/app/models/initiative.js
import Ember from 'ember';
import DS from 'ember-data';

export default DS.Model.extend({
  title:       DS.attr('string'),
  description: DS.attr('string'),
  timePeriod:  DS.attr('string'),
  population:  DS.attr('string'),
  employeeNum: DS.attr('number'),
  countries:   DS.attr('string'),
  language:    DS.attr('string'),
  launchDate:  DS.attr('date'),
  account:     DS.belongsTo('account'),

  fullName: Ember.computed('firstName', 'lastName', function () {
    return `${this.get('firstName')} ${this.get('lastName')}`;
  })
});

File: back_end/portfolio_rails_api/app/models/initiative.rb
class Initiative < ApplicationRecord
  belongs_to :account

  validates :title, presence: true, uniqueness: {scope: :account_id}
  validates :description, length: {maximum: 500}
  validates :time_period, presence: true
  validates :population, presence: true
  validates :employee_num, presence: true, numericality: {greater_than_or_equal_to: 0}
  validates :countries, presence: true
  validates :language, presence: true
  validates :account_id, presence: true
end